src/MicrosoftTeams.ts

```diff
@@ -463,6 +501,43 @@ namespace microsoftTeams
             ensureInitialized(frameContexts.authentication);
 
             sendMessage("authentication.authenticate.failure", reason);
+            window.close();
+        }
+
+        function handleSuccess(result?: string): void
+        {
+            console.log("In handleSuccess");
+            if (authInterval)
+            {
+                clearInterval(authInterval);
+                authInterval = 0;
+            }
+
```

Comment:
Yeah, good point. I was originally thinking that the init would just bubble up to our top-level window and get handled there but that window would have no way of knowing who is calling it or how to give it a proper context. I will also stop the ping loop when the init comes in since at that point it will no longer be necessary.